engines/scumm/imuse_digi/dimuse_triggers.cpp

```diff
@@ -0,0 +1,304 @@
+		ser.syncAsSint32LE(_defers[l].e, VER(103));
+		ser.syncAsSint32LE(_defers[l].f, VER(103));
+		ser.syncAsSint32LE(_defers[l].g, VER(103));
+		ser.syncAsSint32LE(_defers[l].h, VER(103));
+		ser.syncAsSint32LE(_defers[l].i, VER(103));
+		ser.syncAsSint32LE(_defers[l].j, VER(103));
+	}
+
+	if (ser.isLoading())
+		_defersOn = 1;
+}
+
+int IMuseDigiTriggersHandler::setTrigger(int soundId, char *marker, int opcode, int d, int e, int f, int g, int h, int i, int j, int k, int l, int m, int n) {
+	if (soundId == 0) {
+		return -5;
+	}
+
+	if (marker == NULL) {
+		marker = _emptyMarker;
+	}
+
+	if (strlen(marker) >= 256) {
+		debug(5, "IMuseDigiTriggersHandler::setTrigger(): ERROR: attempting to set trigger with oversized marker string");
+		return -5;
+	}
+
+	for (int index = 0; index < MAX_TRIGGERS; index++) {
+		if (_trigs[index].sound == 0) {
+			_trigs[index].sound = soundId;
+			_trigs[index].clearLater = 0;
+			_trigs[index].opcode = opcode;
+			Common::strlcpy(_trigs[index].text, marker, sizeof(_trigs[index].text));
+			_trigs[index].a = d;
+			_trigs[index].b = e;
+			_trigs[index].c = f;
+			_trigs[index].d = g;
+			_trigs[index].e = h;
+			_trigs[index].f = i;
+			_trigs[index].g = j;
+			_trigs[index].h = k;
+			_trigs[index].i = l;
+			_trigs[index].j = m;
+
+			debug(5, "IMuseDigiTriggersHandler::setTrigger(): Successfully set trigger for soundId %d and marker '%s'", soundId, marker);
+			return 0;
+		}
+	}
+	debug(5, "IMuseDigiTriggersHandler::setTrigger(): ERROR: unable to allocate trigger \"%s\" for sound %d, every slot is full", marker, soundId);
+	return -6;
+}
+
+int IMuseDigiTriggersHandler::checkTrigger(int soundId, char *marker, int opcode) {
+	int r = 0;
+	for (int l = 0; l < MAX_TRIGGERS; l++) {
+		if (_trigs[l].sound != 0) {
+			if (soundId == -1 || _trigs[l].sound == soundId) {
+				if (!strcmp(marker, _emptyMarker) || !strcmp(marker, _trigs[l].text)) {
+					if (opcode == -1 || _trigs[l].opcode == opcode)
+						r++;
+				}
+			}
+		}
+	}
+
+	return r;
+}
+
+int IMuseDigiTriggersHandler::clearTrigger(int soundId, char *marker, int opcode) {
+	for (int l = 0; l < MAX_TRIGGERS; l++) {
+		if ((_trigs[l].sound != 0) && (soundId == -1 || _trigs[l].sound == soundId) &&
+			(!strcmp(marker, _emptyMarker) || !strcmp(marker, _trigs[l].text)) &&
+			(opcode == -1 || _trigs[l].opcode == opcode)) {
+
+			if (_midProcessing) {
+				_trigs[l].clearLater = 1;
+			} else {
+				_trigs[l].sound = 0;
+			}
+		}
+	}
+	return 0;
+}
+
+void IMuseDigiTriggersHandler::processTriggers(int soundId, char *marker) {
+	char textBuffer[256];
+	int r;
+	if (strlen(marker) >= 256) {
+		debug(5, "IMuseDigiTriggersHandler::processTriggers(): ERROR: the input marker string is oversized");
+		return;
+	}
+
+	Common::strlcpy(_textBuffer, marker, sizeof(marker));
+	_midProcessing++;
+	for (int l = 0; l < MAX_TRIGGERS; l++) {
+		if (!_trigs[l].sound ||
+			_trigs[l].sound != soundId ||
+			(_trigs[l].text[0] && strcmp(_textBuffer, _trigs[l].text))) {
```

Comment:
I don't like formatting here. maybe one line or add spaces for conditions

Reply:
Hmm, for this particular case a one line doesn't seem so bad, I'll do that